node/src/components/consensus/era_supervisor.rs

```diff
@@ -425,28 +429,24 @@ where
         block_header: BlockHeader,
         responder: Responder<FinalitySignature>,
     ) -> Effects<Event<I>> {
-        // TODO - we should only sign if we're a validator for the given era ID.
-        let signature = asymmetric_key::sign(
-            block_header.hash().inner(),
-            &self.era_supervisor.secret_signing_key,
-            &self.era_supervisor.public_signing_key,
-            self.rng,
-        );
-        let mut effects = responder
-            .respond(FinalitySignature::new(
-                block_header.hash(),
-                signature,
-                self.era_supervisor.public_signing_key,
-            ))
-            .ignore();
-        if block_header.era_id() < self.era_supervisor.current_era {
-            trace!(era_id = %block_header.era_id(), "executed block in old era");
+        let our_pk = self.era_supervisor.public_signing_key;
+        let our_sk = self.era_supervisor.secret_signing_key.clone();
+        let era_id = block_header.era_id();
+        if self.era_supervisor.is_validator_in(&our_pk, era_id) {
+            // TODO: Only sign if a validator!
```

Comment:
Why not move that code here right away?

Reply:
I moved it out at the last moment: I think we're supposed to _always_ respond if we get a responder?
Making this an `Option<_>` also felt wrong and required a lot of changes. So I started wondering whether `HandleLinearBlock` should really be a _request_ at all, or rather an _announcement_…

Reply:
> I moved it out at the last moment: I think we're supposed to always respond if we get a responder?

Oh, you're right. Let's leave it as-is. It probably require more thinking.